Python 3 does *not* expect ASCII from `sys.stdin`. It'll open `stdin` in text mode and make an educated guess as to what encoding is used. That guess may come down to `ASCII`, but that is not a given. See the [`sys.stdin` documentation](http://docs.python.org/3/library/sys.html#sys.stdin) on how the codec is selected.

Like other file objects opened in text mode, the `sys.stdin` object derives from the [`io.TextIOBase` base class](http://docs.python.org/3/library/io.html#io.TextIOBase); it has a `.buffer` attribute pointing to the underlying buffered IO instance (which in turn has a `.raw` attribute).

Wrap the `sys.stdin.buffer` attribute in a new [`io.TextIOWrapper()` instance](http://docs.python.org/3/library/io.html#io.TextIOWrapper) to specify a different encoding:

    import io
    import sys

    input_stream = io.TextIOWrapper(sys.stdin.buffer, encoding='utf-8')

Alternatively, set the [`PYTHONIOENCODING` environment variable](http://docs.python.org/3/using/cmdline.html#envvar-PYTHONIOENCODING) to the desired codec when running python.

From Python 3.7 onwards, you can also [reconfigure the existing `std*` wrappers](https://docs.python.org/3.7/library/io.html#io.TextIOWrapper.reconfigure), provided you do it at the start (before any data has been read):

    # Python 3.7 and newer
    sys.stdin.reconfigure(encoding='utf-8')